conjur/cli.py

```diff
@@ -59,143 +63,154 @@ def copyright():
+                                     help='Enable debugging output',
+                                     action='store_true')
 
+        global_optional.add_argument('--insecure',
+                                     help='Skip verification of server certificate (not recommended for production)',
+                                     dest='ssl_verify',
+                                     action='store_false')
 
         resource, args = Cli._parse_args(parser)
 
-        # We don't have a good "debug" vs "verbose" separation right now
-        if args.verbose is True:
-            args.debug = True
-
         # TODO Add tests for exception handling logic
+        # # pylint: disable=broad-except
         try:
+            self._setup_logging(args.debug)
+
             Cli.run_client_action(resource, args)
         except FileNotFoundError as not_found_error:
             sys.stdout.write(f"Error: No such file or directory: '{not_found_error.filename}'")
-            logging.info('Error: No such file or directory: %s', not_found_error.filename)
-            sys.exit(1)
         except requests.exceptions.HTTPError as http_error:
             sys.stdout.write(str(http_error))
-            logging.info(str(http_error))
-            sys.exit(1)
-        else:
-            sys.exit(0)
+        except ValueError as value_error:
+            sys.stdout.write(str(value_error))
+        except RuntimeError as runtime_error:
+            sys.stdout.write(str(runtime_error))
+        except Exception as general_exception:
+            sys.stdout.write(str(general_exception))
+
+        sys.exit(0)
 
     @staticmethod
     def run_client_action(resource, args):
         """
         Helper for creating the Client instance and invoking the appropriate
         api class method with the specified parameters.
         """
+        if resource == 'init':
+            init_input_obj = InputConjurrc(args.url,
+                                           args.account,
+                                           args.cert)
+            init_logic = InitLogic()
+
+            input_controller = InitController(init_input_obj,
+                                              init_logic,
+                                              args.force)
+            input_controller.call()
+            sys.exit(0)
 
-        ca_bundle = None
-        if args.ca_bundle:
-            ca_bundle = os.path.expanduser(args.ca_bundle)
-
-        # We want explicit definition of things to pass into the client
-        # to avoid ambiguity
-        client = Client(url=args.url,
-                        account=args.account,
-                        api_key=args.api_key,
-                        login_id=args.login_id,
-                        password=args.password,
-                        ssl_verify=args.ssl_verify,
-                        ca_bundle=ca_bundle,
-                        debug=args.debug)
+        client = Client(ssl_verify=args.ssl_verify)
```

Comment:
This does two things:
- Completely breaks API compatibility with older client lib versions
- Fails to isolate components properly

The first one is a serious problem that can be worked around but Client class should not have any notion as to how to get the client config values (unless we use a helper class as a parameter)

Reply:
Understood and good catch. There is a requirement to remove the ability to pass params via command line to configure a connection to Conjur so shouldn't this be transferred over to the client as well. How would you suggest we meet the requirement but aim for backwards capability? 
The client is initialized from the `cli.py` which accepts user input.  Would you suggest I initialize the client in cli.py setting all options to None like so:

```
# cli.py
client = Client(url=args.url,
     account=None,
     api_key=None,
     login_id=None,
     password=None,
     ssl_verify=None,
     ca_bundle=None,
     debug=args.debug)
```
and the same for the client
```
# client.py
def __init__(self,
                 account=None,
                 api_key=None,
                 ca_bundle=None,
                 debug=False,
                 http_debug=False,
                 login_id=None,
                 password=None,
                 ssl_verify=True,
                 url=None):
```
and add back the logic for using the api_key / password to interact with Conjur?

Reply:
> There is a requirement to remove the ability to pass params via command line to configure a connection to Conjur

This seems like a really bad idea - without this capability, automation cannot be used in conjunction with the CLI

> How would you suggest we meet the requirement but aim for backwards capability?

There is no reason to remove the old usage at all as all of the params are optional when invoked.

Reply:
I updated the integration tests to take from a generated conjurrc file instead of via command line args as they originally built to do. I pushed to GH. Let me know what you think 

Old usage being how the client class originally was? If so, I will revert my changes to the client class

Reply:
I reverted changes I made to the Client class to avoid breakage. Take a look 🙏